Source/Core/Quaternion.js

```diff
@@ -184,18 +191,30 @@ define([
      * @param {Quaternion} [result] The object onto which to store the result.
      * @returns {Quaternion} The modified result parameter or a new Quaternion instance if none was provided.
      */
-    Quaternion.fromHeadingPitchRoll = function(heading, pitch, roll, result) {
+    Quaternion.fromHeadingPitchRoll = function(headingOrHeadingPitchRoll, pitchOrResult, roll, result) {
         //>>includeStart('debug', pragmas.debug);
-        Check.typeOf.number('heading', heading);
-        Check.typeOf.number('pitch', pitch);
-        Check.typeOf.number('roll', roll);
+        if (headingOrHeadingPitchRoll instanceof HeadingPitchRoll) {
+          Check.typeOf.object('headingPitchRoll',headingOrHeadingPitchRoll );
```

Comment:
There are still whitespace issues. Four spaces should be used for each indent. Please update throughout.